File: LeetCode/src/main/java/com/jack/queue/LeetCode933.py
from collections import deque


class RecentCounter(object):

    def __init__(self):
        self.q = deque()
   
    def ping(self, t):
        self.q.append(t)
        while self.q[0]< t-3000:
            self.q.popleft()
            
        return len(self.q)




if __name__ == "__main__":
    pass

File: LeetCode/src/main/java/com/jack/queue/LeetCode933.java
package com.jack.queue;

import java.util.ArrayDeque;
import java.util.Deque;

public class LeetCode933 {
    public static void main(String[] args) {
        
    }

    static class RecentCounter {

        private Deque<Integer> queue = new ArrayDeque<>();

        public RecentCounter() {
            
        }
        
        public int ping(int t) {
            queue.offer(t);

            while (queue.peekFirst()< t-3000){
                queue.pollFirst();
            }

            return queue.size();
            
        }
}

}
